### PR TITLE
Diagnostic Output for AFMV

### PR DESCRIPTION
**Pull request by code-Gambler**
This PR closes the this Issue.
#17 

It adds a test gimple pass which prints "Hello World" for now, but In the future I would like to integrate it with cloning and pruning and provide some diagnostic output to the developer .

### CODE REVIEW

File: gcc/test.cc [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    /* todo_flags_start */
+    0 /* todo_flags_finish */
+  };
+
+  /* Test pass for dumping a test message. */
+  class pass_test_dump: public gimple_opt_pass {
+    public: pass_test_dump(gcc::context * ctxt): gimple_opt_pass(pass_data_gimple_test, ctxt) {}
+
+    /* Pass gate function. */
+    bool gate(function*) final override {
```
Comment (MatthewDeNapoli) [inline gcc/test.cc:34]: move the return type to the line above

File: gcc/test.cc [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      return true;
+    }
+
+    /* Pass execution function. */
+    unsigned int execute(function*) final override {
```
Comment (MatthewDeNapoli) [inline gcc/test.cc:39]: move the return type to the line above

File: gcc/test.cc [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      fprintf(stderr, "Hello World\n");
+      return 0;
+    }
+  };
+}
+
+/* Create the test dump pass. */
+gimple_opt_pass * make_pass_test_dump(gcc::context * ctxt) {
```
Comment (MatthewDeNapoli) [inline gcc/test.cc:47]: move the return type to the line above

**Review by MatthewDeNapoli — CHANGES_REQUESTED** [review]

**Comment by code-Gambler** [comment]
Changes Done!

**Review by MatthewDeNapoli — APPROVED** [review]

**Pull request merged**